lite.js

```diff
@@ -3,194 +3,74 @@
+const SimpleHandler = require('./lib/define-property-handler')
 
 class SimpleRDF {
   constructor (context, iri, graph) {
-    this._iri = buildIri(iri)
-    this._objects = {}
+    this._core = new SimpleCore(this)
+    this._handler = new SimpleHandler(this, this._core)
+
+    this._core.iri = SimpleCore.buildIri(iri)
 
     this.context(context)
     this.graph(graph || rdf.createGraph())
   }
 
   toString () {
-    return this._graph.toString()
+    return this._core.graph.toString()
   }
 
   context (context) {
     if (context) {
-      this._context = context instanceof SimpleContext ? context : new SimpleContext(context)
+      this._core.context = context instanceof SimpleContext ? context : new SimpleContext(context)
 
-      this._context.descriptions().forEach((description) => {
+      this._core.context.descriptions().forEach((description) => {
         // access values with full IRI
-        addProperty.call(
-          this,
-          description.predicate,
-          description.predicate,
-          description.options)
+        this._handler.addProperty(description.predicate, description.predicate, description.options)
 
         // access values with short property
-        addProperty.call(
-          this,
-          description.property,
-          description.predicate,
-          description.options)
+        this._handler.addProperty(description.property, description.predicate, description.options)
       })
     }
 
-    return this._context
+    return this._core.context
   }
 
   iri (iri) {
     if (iri) {
-      iri = buildIri(iri)
+      iri = SimpleCore.buildIri(iri)
 
-      updateSubject(this._graph, this._iri, iri)
-      updateObject(this._graph, this._iri, iri)
+      if (!iri.equals(this._core.iri)) {
+        this._core.updateSubject(iri)
+        this._core.updateObject(iri)
 
-      this._iri = iri
+        this._core.iri = iri
+      }
     }
 
-    return this._iri
+    return this._core.iri
   }
 
   graph (graph) {
     if (graph) {
-      this._graph = graph
+      this._core.graph = graph
 
-      this._graph.match(this._iri).forEach((triple) => {
+      this._core.graph.match(this._core.iri).forEach((triple) => {
         let predicate = triple.predicate.toString()
-        let descriptor = Object.getOwnPropertyDescriptor(this, predicate)
 
-        if (!descriptor) {
-          addProperty.call(this, triple.predicate.toString(), predicate)
+        if (!this._handler.hasProperty(predicate)) {
+          this._handler.addProperty(triple.predicate.toString(), predicate)
         }
       })
     }
 
-    return this._graph
+    return this._core.graph
   }
 
   child (iri) {
-    return new SimpleRDF(this._context, iri, this._graph)
-  }
-}
-
-function buildIri (iri) {
-  if (typeof iri === 'string') {
-    return rdf.createNamedNode(iri)
-  } else {
-    return iri || rdf.createBlankNode()
-  }
-}
-
-function updateSubject (graph, oldSubject, newSubject) {
```

Comment:
`buildIri` was moved to a static method. `updateSubject` and `updateObject` was always called with `this._graph` and `this._iri`. It's a little bit hidden, but from the logic point of view it was already a method.